src/api/test/activexml_matcher.rb

```diff
@@ -175,7 +175,10 @@ def self.match(node, conditions)
     if conditions[:ancestor]
       return false unless catch :found do
         p = node
-        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
+        while p
+          p = p.parent
```

Comment:
On 2015-09-16 04:23:44 -0700, Moisés Déniz Alemán wrote:
> > @@ -175,7 +175,10 @@ def self.match(node, conditions)
> >      if conditions[:ancestor]
> >        return false unless catch :found do
> >          p = node
> > -        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
> > +        while p
> > +          p = p.parent
> 
> Yes. @marcus-h commented it here https://github.com/openSUSE/open-build-service/pull/1132#issuecomment-140434914
> The first value of *p* should be an ancestor, not the node itself... isn't it?
> 
Yep (that's what the old code did/does). It could be translated as

p = node.parent
while p
  throw :found, true if match(p, conditions[:ancestor])
  p = p.parent
end